src/sql/workbench/parts/objectExplorer/common/objectExplorerViewTreeShim.ts

```diff
@@ -154,6 +156,30 @@ export class OEShimService extends Disposable implements IOEShimService {
 			}
 		}
 		icon = icon.toLowerCase();
+		// Change the database if the node has a different database
+		// than its parent
+		let updatedPayload: IConnectionProfile = undefined;
+		if (node.nodeTypeId === NodeType.Database) {
+			const database = node.getDatabaseName();
+			if (database) {
+				updatedPayload = {
+					connectionName: parentNode.payload.connectionName,
+					serverName: parentNode.payload.serverName,
+					databaseName: node.getDatabaseName(),
+					userName: parentNode.payload.userName,
+					password: parentNode.payload.password,
+					authenticationType: parentNode.payload.authenticationType,
+					savePassword: parentNode.payload.savePassword,
+					groupFullName: parentNode.payload.groupFullName,
+					groupId: parentNode.payload.groupId,
+					providerName: parentNode.payload.providerName,
+					saveProfile: false,
+					id: parentNode.payload.id,
+					azureTenantId: parentNode.payload.azureTenantId,
+					options: parentNode.payload.options
+				};
```

Comment:
So if we add an optional var to this interface this is going to silently drop that var. Object.assign is something we use elsewhere but that has issues too since it's basically an incomplete shallow copy...this is a general question for the API though (we should probably provide a deepClone function) but for now you should at the very least use Object.assign and then open an issue for further discussion

Reply:
This interface will never have an optional var because it's coming from the extension payload, which is fixed for a tree node. But if you really insist on changing this, I'll do it.

Reply:
The situation I'm saying we need to be careful of is let's say we change IConnectionProfile and add a new optional var - such as you're doing here. In that case the file would compile (because it's optional) even though you now wouldn't be copying that value over to the new updated payload and so would lose whatever that value was.